File: tests/rspec/pages/utilities/debug_extensions.rb
class DebugExtensions < ControlPanelPage
	set_url_matcher /utilities\/extensions/

	element :heading, 'div.w-12 div.box h1'

	element :no_results, 'div.box div.tbl-ctrls form div.tbl-wrap table tr.no-results'

	element :addon_name_header, 'div.box div.tbl-ctrls form div.tbl-wrap table tr th:first-child'
	element :status_header, 'div.box div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(2)'
	element :manage_header, 'div.box div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(3)'
	element :checkbox_header, 'div.box div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(4)'

	elements :addons, 'div.box div.tbl-ctrls form div.tbl-wrap table tr'
	elements :addon_names, 'div.box div.tbl-ctrls form div.tbl-wrap table tr td:first-child'
	elements :statuses, 'div.box div.tbl-ctrls form div.tbl-wrap table tr td:nth-child(2)'

	element :bulk_action, 'form fieldset.tbl-bulk-act select[name="bulk_action"]', visible: false
	element :action_submit_button, 'form fieldset.tbl-bulk-act input.submit'

	def load
		self.open_dev_menu
		click_link 'Utilities'
		click_link 'Debug Extensions'
	end
end

File: tests/cypress/cypress/elements/pages/utilities/DebugExtensions.js
import ControlPanel from '../ControlPanel'

class DebugExtensions extends ControlPanel {
    constructor() {
        super()
        this.url = 'admin.php?/cp/utilities/extensions';


        this.selectors = Object.assign(this.selectors, {
        	  "heading" : '.ee-main .title-bar .title-bar__title',
        	  "no_results" : 'form div.table-responsive table tr.no-results',
			  "addon_name_header" : 'form div.table-responsive table tr th:first-child',
			  "status_header" : 'form div.table-responsive table tr th:nth-child(2)',
			  "manage_header" : 'form div.table-responsive table tr th:nth-child(3)',
			  "checkbox_header" : 'form div.table-responsive table tr th:nth-child(4)',
			  "addons" : 'form div.table-responsive table tr',
			  "addon_names" : 'form div.table-responsive table tr td:first-child',
			  "statuses" : 'form div.table-responsive table tr td:nth-child(2)'
        })

    }

    load(){
		this.open_dev_menu()
    	cy.get('body > .dropdown').find('a:contains("Utilities")').click()
    	this.get('wrap').find('a:contains("DebugExtensions")').click()
    }

}
export default DebugExtensions;

